tests/common/gcp_api/admin_directory_test.py

```diff
@@ -80,6 +80,20 @@ def test_get_groups_raises(self):
         with self.assertRaises(api_errors.ApiExecutionError):
             self.ad_api_client.get_groups()
 
+    @mock.patch.object(admin, 'LOGGER')
+    @mock.patch.object(admin, 'api_helpers')
+    def test_get_groups_raises_http_access_token_refresh_error(self, mock_api_helper, mock_logger):
+        mock_api_helper.flatten_list_results.side_effect = client.HttpAccessTokenRefreshError
+
+        with self.assertRaises(client.HttpAccessTokenRefreshError):
+            self.ad_api_client.get_groups()
+        mock_logger.error.assert_called_with(
+            'Failed to retrieve group data due to authentication failure. '
```

Comment:
If you want to re-use the message, do you want to make it a constant in the app code, and then just import it here for re-use?